src/chapters/aspdotnet-model-classes/data-layer.rst

```diff
@@ -0,0 +1,284 @@
+.. _data-layer:
+
+Models and Data 
+===============
+
+In order to work with data, we need to add another element to our MVC application. Say for example,
+we want to do things like remove an event from our list. Well, if two events both have the same name, 
+how might we identify which of those items to delete? We can't yet. So we need to tweak how we store
+event data. 
+
+In ``CodingEvents``, we add a unique identifier field to ``Events`` to better handle and track distinct 
+``Event`` instances. We'll also create another model class called ``EventData`` to encapsulate data storage 
+and prepare ourselves for decoupling data from controller classes.
+
+Add a Unique Id - Video
+-----------------------
+
+.. TODO: Add create uid in model class video
+.. topics covered: create id field on event class
+
+YOUTUBE VIDEO HERE
+
+.. admonition:: Note
+
+   The starter code for this video is found at the `add-property branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/add-property>`__
+   of ``CodingEventsDemo``. The final code presented in this 
+   video is found on the `add-id branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/add-id>`__.
+   As always, code along to the videos on your own ``CodingEvents`` project.
+
+.. index:: ! uid, ! unique identifier
+
+Add a Unique Id - Text
+----------------------
+
+Identifying data by a user-defined string called ``name`` is not a sustainable or scalable method
+of handling data in most situations. Consider the address book example. How can
+we distinguish between two contact entries with the same name field? It is a frequent
+practice to add a **unique identifier** field (sometimes called, or even labelled, **uid**) to a class 
+responsible for modelling data. This ensures that our address book can contain two separate entries for 
+our contacts who have the same name as one another. 
```

Comment:
See the above comment. I would keep this in the context of events and ``CodingEvents``.

Reply:
see the comment above - i don't think it would be harmful to add other contexts to why adding ids are good. we dont want to get stuck in the world of coding events, but instead reinforce that coding events is one example of how to implement these concepts

Reply:
Gotcha! I can see how this would work with the video accompanying it now!